FrostCord/FrostCord
# Mini Profiles

# DO NOT MERGE UNTIL #32 IS MERGED
## Mini Profiles
This PR introduces the beginning of profiles via mini-profiles on hover. Likewise, this PR allows nicknames to work for users

## API Changes
- Adds mini profiles which show up on hover
- Adds support for nicknames on the frontend
- Adds richer role integration to chat/profiles (chat message headers now use the highest role the user has that has a non-null color and mini-profiles list the roles a user has)
- Cleaned up APIs for fetching messages w/ profiles using a postgres function, amended types in TS accordingly

## Testing
- Verify mini profiles with a small text field appear when you hover over a user's username in chat
- Verify the mini profile accurately reflects a user's nickname as set in the server_users table for the server
- Verify after switching channels, or WITHOUT switching channels, have another user send a message, then send a message from the user who's nickname was changed. Their message header (pfp + name) should now reflect the new settings

## Known Issues
- Hovering to view a miniprofile you've already seen before performs additional renders, investigation should be done to cut down on renders. **UPDATE** - Cause is `react-tooltip`, standalone this renders as expected. Switch this to a custom tooltip hover component when it's ready.
